packages/sdk-react/src/components/userRoles/UserRoleDetails/helpers.ts

```diff
@@ -65,8 +63,27 @@ const transformActionsToComponentFormat = <T extends string, P>(
   return Object.assign({}, permission, ...permissionsObject);
 };
 
+const createInitialPermissionState = <
+  T extends
+    | components['schemas']['ActionEnum']
+    | components['schemas']['PayableActionEnum']
+>(
+  objectType: payablePermissionsObjectType | commonPermissionsObjectType,
+  actionEnum: T[]
+): PayablePermissionRow | CommonPermissionRow => {
+  let permission: Record<string, boolean | typeof objectType> = {};
+
+  permission.name = objectType;
+
+  actionEnum.forEach((action) => {
+    permission[action] = false;
+  });
+
+  return permission as PayablePermissionRow | CommonPermissionRow;
```

Comment:
I see a complicated union, but, seems, output generic is needed. However, if typing is too complicated, this is fine.

My only suggestion would be: that's fine to do `as T` for generics resolution.

In my case, if I  need to apply `as` to something that definitely could be determined using runtime checks, but I didn't do this - here is the point where TypeScript reminds me that, probably, the initial implementation was wrong.

Reply:
Yes, the types of roles implementation are overcomplicated. I realized it again when I was moving it to Qraft.
I created a task in our tech debt epic.